lib/createDuck.ts

```diff
@@ -13,6 +13,11 @@ export interface Duck<TState, TPayload = any> {
     (payload?: TPayload): Action<TPayload>;
     actionType: string;
     payloadReducer: PayloadReducer<TState, TPayload>;
+    /**
+     * This variable is never used.
+     * It is provided to allow for example "type MyAction = typeof myDuck.ActionTypeDef;"
+     */
+    ActionTypeDef: Action<TPayload>;
```

Comment:
Yeah I thought that - however it fixes a huge problem when trying to write proper reducers / handler code for Redux with typescript.
And the good side is, this field is only by declaration - i.e. it won't show up anywhere else except in typescript declarations.

The only other solution is using something like https://github.com/piotrwitek/utility-types#call which however creates runtime code (as there is no other way for that in typescript yet) or to wait for https://github.com/Microsoft/TypeScript/issues/6606 to be accepted which could still take a while.

Until that happened I thought this might be a nice add-in and if you want I could show some samples where this is a great help in cases that could not be solved without it.